driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -617,10 +617,6 @@ private static void updateInfo(
       // want to setup a different port for SSL and non-SSL conns).
       InetAddress nativeAddress = row.getInet("native_transport_address");
       int nativePort = row.getInt("native_transport_port");
-      if (cluster.getCluster().getConfiguration().getProtocolOptions().getSSLOptions() != null
-          && !row.isNull("native_transport_port_ssl")) {
-        nativePort = row.getInt("native_transport_port_ssl");
-      }
```

Comment:
This code added for JAVA-2967 ends up overwriting the `native_transport_port` for the initial control connection before it gets stored in Metadata#hosts. When we later receive a `REMOVE_NODE` event for this host we are unable to find the entry to remove because the broadcast-rpc port does not match. This results in the driver skipping the graceful connection closing code and continuing to use the connection until it's suddenly closed serverside by the operator killing the coordinator pod.

In it's current state, removing the use of `native_transport_port_ssl` column probably breaks some DSE stuff. We should revisit that earlier patch to figure out a solution that works for both backends.